hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/slowlog/TestSlowLogRecorder.java

```diff
@@ -0,0 +1,266 @@
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestSlowLogRecorder.class);
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestSlowLogRecorder.class);
+
+  private static final HBaseTestingUtility HBASE_TESTING_UTILITY = new HBaseTestingUtility();
+
+  private SlowLogRecorder slowLogRecorder;
+
+  private static Configuration applySlowLogRecorderConf(int eventSize) {
+    Configuration conf = HBASE_TESTING_UTILITY.getConfiguration();
+    conf.setBoolean(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY, true);
+    conf.setInt(HConstants.SLOW_LOG_RING_BUFFER_SIZE, eventSize);
+    return conf;
+  }
+
+  private SlowLogPayload getSlowLogPayload(int i) {
+    SlowLogPayload slowLogPayload = SlowLogPayload.newBuilder()
+      .setCallDetails("call_" + i)
+      .setClientAddress("client_" + i)
+      .setMethodName("method_" + i)
+      .setUserName("userName_" + i)
+      .setMultiGets(i)
+      .setMultiMutations(i)
+      .setMultiServiceCalls(i)
+      .setProcessingTime(i + 500)
+      .setQueueTime(i + 400)
+      .setResponseSize(i + 200)
+      .setStartTime(EnvironmentEdgeManager.currentTime())
+      .setServerClass("class_" + i)
+      .setParam("param_" + i)
+      .build();
+    return slowLogPayload;
+  }
+
+  /**
+   * confirm that for a ringbuffer of slow logs, payload on given index of buffer
+   * has expected elements
+   *
+   * @param i index of ringbuffer logs
+   * @param j data value that was put on index i
+   * @param slowLogPayloads list of payload retrieved from {@link SlowLogRecorder}
+   */
+  private void confirmPayloadParams(int i, int j, List<SlowLogPayload> slowLogPayloads) {
+    Assert.assertEquals(slowLogPayloads.get(i).getClientAddress(), "client_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getCallDetails(), "call_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getMethodName(), "method_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getUserName(), "userName_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getProcessingTime(), j + 500);
+    Assert.assertEquals(slowLogPayloads.get(i).getQueueTime(), j + 400);
+    Assert.assertEquals(slowLogPayloads.get(i).getResponseSize(), j + 200);
+    Assert.assertEquals(slowLogPayloads.get(i).getServerClass(), "class_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getParam(), "param_" + j);
+  }
+
+  @Test
+  public void testOnlieSlowLogConsumption() throws Exception {
+    Configuration conf = applySlowLogRecorderConf(8);
+    slowLogRecorder = new SlowLogRecorder(conf);
+    Assert.assertEquals(slowLogRecorder.getSlowLogPayloads().size(), 0);
+    LOG.debug("Initially ringbuffer of Slow Log records is empty");
+
+    int i = 0;
+
+    // add 5 records initially
+    for (; i < 5; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      slowLogRecorder.addSlowLogPayload(slowLogPayload);
+    }
+
+    HBASE_TESTING_UTILITY.waitFor(60000,
```

Comment:
Members of the `MediumTest` category: the whole class is supposed to complete within 50secs (see the Javadoc). 60 seconds blows your time allotment. Either reduce the wait or bump the test category to Large.

Same with the other invocations below.